### PR TITLE
Feature/lvpytcdxii 19

### PR DESCRIPTION
**Pull request by ement06**
* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=1032)

## Code reviewers

- [ ] @sikyrynskiy
- [ ] @olya_petryshyn
- [ ] @taraskonchak
- [ ] @OlyaKh00
- [ ] @ement06
- [ ] @iPavliv
- [ ] @khadai
- [ ] @Anastasia_Siromska
- [ ] @romichh

### Second Level Review

- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: groups_service/models/group.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    String,
+    DateTime
+)
+from groups_service.db import DB
+
+
+
+class GroupsModel(DB.Model): # pylint: disable=too-few-public-methods
+    """Implementation of Group entity."""
+    id = Column(Integer(), primary_key=True) # pylint: disable=redefined-builtin, redefined-builtin, invalid-name
```
Comment (iPavliv) [inline groups_service/models/group.py]: change to group_id and delete pylint disable comment

File: .gitignore [hunk 2]
```diff
@@ -102,3 +102,6 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+# vscode
+.vscode
```
Comment (LSikyr) [inline .gitignore]: No empty line in the end of the file. Very disappointed in you!!!

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+prod-env:
+	 make install; \
+	 make set-path; \
+	 export FLASK_ENV="production"; \
+	 flask run --port=5052;
+
+lint:
+	pylint setup.py form_service/
```
Comment (LSikyr) [inline Makefile]: Change form_service to group_service!!!!!!

File: groups_service/migrations/README [hunk 4]
```diff
@@ -0,0 +1 @@
+Generic single-database configuration.
```
Comment (LSikyr) [inline groups_service/migrations/README]: No blank line!!!!!!!!!!!!

File: groups_service/models/group.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+"""Model for group service."""
+import datetime
+from sqlalchemy import (
+    Column,
+    Integer,
+    String,
+    DateTime
+)
+from groups_service.db import DB
```
Comment (LSikyr) [inline groups_service/models/group.py:9]: imports are not in alpabetic order

File: groups_service/views/groups.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+@APP.route('/api/groups/<owner_id>', methods=['GET'])
+def get_groups_owner(owner_id):
+    '''Get method for all groups by owner_id'''
+    if owner_id.isnumeric():
+        groups = GroupsModel.query.filter_by(owner_id=owner_id)
+        for group in groups:
+            group.members = list(map(int, group.members.split(',')))
+        return GROUPS_SCHEMA.jsonify(groups)
+    return jsonify({'message': 'Not correct URL'})
```
Comment (LSikyr) [inline groups_service/views/groups.py]: No need to jsonify dict!!!!

File: groups_service/views/groups.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    return jsonify(result.data)
+
+@APP.route('/api/group/<group_id>')
+def get_group(group_id):
+    """Get method for one group by group_id."""
+    if group_id.isnumeric():
+        group = GroupsModel.query.get_or_404(group_id)
+        group.members = list(map(int, group.members.split(',')))
+        return GROUP_SCHEMA.jsonify(group), 200
+    return jsonify({'message': 'Not correct URL'}), 400
```
Comment (LSikyr) [inline groups_service/views/groups.py]: multiple returns!!!!

Comment (LSikyr) [inline groups_service/views/groups.py]: No need to jsonify dict!!!!

File: groups_service/views/groups.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    DB.session.add(new_group)
+    DB.session.commit()
+    return GROUP_SCHEMA.jsonify(new_group), 201
+
+@APP.route('/api/groups', methods=['GET'])
+def all_groups():
+    """Get method for all groups."""
+    groups = GroupsModel.query.all()
+    result = GROUPS_SCHEMA.dump(groups)
+    return jsonify(result.data)
```
Comment (LSikyr) [inline groups_service/views/groups.py]: GROUPS_SCHEMA.dump(groups).data
return result

File: groups_service/views/groups.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        owner_id=owner_id,
+        members=members
+        ).first())
+    if exists_group:
+        return jsonify({'message': 'Group alredy exists'}), 400
+
+    new_group = GroupsModel(title, form_id, owner_id, members)
+    DB.session.add(new_group)
+    DB.session.commit()
+    return GROUP_SCHEMA.jsonify(new_group), 201
```
Comment (LSikyr) [inline groups_service/views/groups.py]: multiple returns

File: groups_service/__init__.py [hunk 10]
```diff
@@ -1,4 +1,13 @@
 """Init users service"""
```
Comment (LSikyr) [inline groups_service/__init__.py]: Copy paste detected!!!

**Review by LSikyr — CHANGES_REQUESTED** [review]

**Review by LSikyr — APPROVED** [review]

**Review by olyapetryshyn — APPROVED** [review]

**Pull request merged**